pombola/south_africa/views.py

```diff
@@ -1962,3 +1962,101 @@ def get_context_data(self, **kwargs):
             context['data'] = None
 
         return context
+
+class SAMpAttendanceView(TemplateView):
+    template_name = "south_africa/mp_attendance.html"
+
+    def calculate_abs_percenatge(self, num, total):
+        """
+        Return the truncated value of num, as a percentage of total.
+        """
+        return int("%.*f" % (0, (num / total * 100)))
+
+    # def post(self, request, *args, **kwargs):
+    #     context = self.get_context_data()
+    #     return super(SAMpAttendanceView, self).render_to_response(context)
+
+    def get_context_data(self, **kwargs):
+        context = {}
+        api_url = 'https://api.pmg.org.za/committee-meeting-attendance/summary/'
+
+        #  A:   Absent
+        #  AP:  Absent with Apologies
+        #  DE:  Departed Early
+        #  L:   Arrived Late
+        #  LDE: Arrived Late and Departed Early
+        #  P:   Present
+
+        present_codes = ['P', 'L', 'LDE', 'DE']
+        arrive_late_codes = ['L', 'LDE']
+        depart_early_codes = ['DE', 'LDE']
+
+        resp = requests.get(api_url)
+        data = json.loads(resp.content)
+
+        context['year'] = str(
+            dateutil.parser.parse(data['results'][0]['end_date']).year)
+        context['party'] = ''
+
+        for key in ('year', 'party'):
+            if key in self.request.GET:
+                context[key] = self.request.GET[key]
+
+        context['attendance_data'] = []
+        context['years'] = []
+        context['download_url'] = 'http://api.pmg.org.za/committee-meeting-attendance/data.xlsx'
+
+        for annual_attendance in data['results']:
+            year = str(dateutil.parser.parse(annual_attendance['end_date']).year)
+            context['years'].append(year)
+
+            if year == context['year']:
+                parties = set(ma['member']['party_name'] for
+                    ma in annual_attendance['attendance_summary'])
+                parties.discard(None)
+                context['parties'] = sorted([p for p in parties])
+
+                attendance_summary = annual_attendance['attendance_summary']
+                if context['party']:
+                    attendance_summary = [mbr_att for mbr_att in attendance_summary if
+                        mbr_att['member']['party_name'] == context['party']]
+
+                aggregate_total = aggregate_present = 0
+
+                for summary in attendance_summary:
+                    total = sum([v for v in summary['attendance'].itervalues()])
```

Comment:
It's a tiny and nearly insignificant thing here, but these could just be generator expressions rather than list comprehensions that are passed to `sum` to save on constructing a list, e.g.

    sum(v for v in summary['attendance'].itervalues())